### PR TITLE
Key UX improvements

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/web/eth_keys_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	resp, cleanup := client.Get("/v2/keys/eth")
+	defer cleanup()
+
+	balances := []presenters.ETHKey{}
+	err = cltest.ParseJSONAPIResponse(t, resp, &balances)
+	assert.NoError(t, err)
+
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	assert.Len(t, balances, 0)
+	// kst.AssertExpectations(t)
```
Comment (RyanRHall) [inline core/web/eth_keys_controller_test.go:93]: :scissors:

File: core/web/eth_keys_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+	require.NoError(t, app.StartAndConnect())
+
+	resp, cleanup := client.Post("/v2/keys/eth", nil)
+	defer cleanup()
+
+	cltest.AssertServerResponse(t, resp, 201)
+
+	ethClient.AssertExpectations(t)
+}
```
Comment (RyanRHall) [inline core/web/eth_keys_controller_test.go:122]: There are no Import / Export tests for EthKeysController

File: core/store/store_test.go [hunk 3]
```diff
@@ -170,3 +176,121 @@ func TestStore_SyncDiskKeyStoreToDB_DBKeyAlreadyExists(t *testing.T) {
...
+	keys, err = store.SendKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 0)
+
+	keys, err = store.AllKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 0)
+}
+
+func TestKeyStore_Import(t *testing.T) {
```
Comment (RyanRHall) [inline core/store/store_test.go:241]: nitpick - some of these tests are named `TestStore_<Action>Key` and some are names `TestKeyStore_<Action>`

File: core/store/key_store.go [hunk 4]
```diff
@@ -4,32 +4,35 @@ import (
...
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
 // EthereumMessageHashPrefix is a Geth-originating message prefix that seeks to
 // prevent arbitrary message data to be representable as a valid Ethereum transaction
 // For more information, see: https://github.com/ethereum/go-ethereum/issues/3731
 const EthereumMessageHashPrefix = "\x19Ethereum Signed Message:\n32"
 
+var ErrKeyStoreLocked = errors.New("keystore is locked (HINT: did you forget to call keystore.Unlock?)")
```
Comment (RyanRHall) [inline core/store/key_store.go:23]: :+1: all error messages should contain the word hint

File: core/services/validators_test.go [hunk 5]
```diff
@@ -309,12 +309,12 @@ func TestValidateServiceAgreement(t *testing.T) {
...
 	store, cleanup := cltest.NewStore(t)
-	_, err := store.KeyStore.NewAccount("password") // matches correct_password.txt
-	assert.NoError(t, err)
-	err = store.KeyStore.Unlock("password")
+	defer cleanup()
+	err := store.KeyStore.Unlock("password")
 	_, fromAddress := cltest.MustAddRandomKeyToKeystore(t, store, 0)
 	assert.NoError(t, err)
-	defer cleanup()
+	_, err = store.KeyStore.NewAccount() // matches correct_password.txt
```
Comment (RyanRHall) [inline core/services/validators_test.go:316]: :scissors: the comment

File: core/services/validators_test.go [hunk 6]
```diff
@@ -309,12 +309,12 @@ func TestValidateServiceAgreement(t *testing.T) {
 	t.Parallel()
 
 	store, cleanup := cltest.NewStore(t)
-	_, err := store.KeyStore.NewAccount("password") // matches correct_password.txt
-	assert.NoError(t, err)
-	err = store.KeyStore.Unlock("password")
+	defer cleanup()
+	err := store.KeyStore.Unlock("password")
```
Comment (RyanRHall) [inline core/services/validators_test.go:313]: use `cltest.Password`

File: core/cmd/remote_client_test.go [hunk 7]
```diff
@@ -815,16 +814,19 @@ func TestClient_IndexTxAttempts(t *testing.T) {
...
-	kst := app.Store.KeyStore.(*mocks.KeyStoreInterface)
+	ethClient := new(mocks.Client)
+
+	app, cleanup := cltest.NewApplicationWithKey(t, ethClient)
 	defer cleanup()
+	verify := cltest.MockApplicationEthCalls(t, app, ethClient)
+	defer verify()
+
+	ethClient.On("BalanceAt", mock.Anything, mock.Anything, mock.Anything).Maybe().Return(big.NewInt(42), nil)
+	ethClient.On("GetLINKBalance", mock.Anything, mock.Anything, mock.Anything).Maybe().Return(assets.NewLink(42), nil)
```
Comment (RyanRHall) [inline core/cmd/remote_client_test.go:828]: are these supposed to be `Maybe()`s ?

File: core/cmd/local_client_test.go [hunk 8]
```diff
@@ -187,7 +187,7 @@ func TestClient_RunNode_CreateFundingKeyIfNotExists(t *testing.T) {
 	ethClient.On("Dial", mock.Anything).Return(nil)
 	store.EthClient = ethClient
 
-	_, err = store.KeyStore.NewAccount("password") // matches correct_password.txt
+	_, err = store.KeyStore.NewAccount() // matches correct_password.txt
```
Comment (RyanRHall) [inline core/cmd/local_client_test.go:190]: :scissors: the comment

File: core/cmd/local_client_test.go [hunk 9]
```diff
@@ -131,7 +131,7 @@ func TestClient_RunNodeWithPasswords(t *testing.T) {
 			ethClient.On("BalanceAt", mock.Anything, mock.Anything, mock.Anything).Return(big.NewInt(10), nil)
 			store.EthClient = ethClient
 
-			_, err = store.KeyStore.NewAccount("password") // matches correct_password.txt
+			_, err = store.KeyStore.NewAccount() // matches correct_password.txt
```
Comment (RyanRHall) [inline core/cmd/local_client_test.go:134]: :scissors: the comment

**Review by RyanRHall — DISMISSED** [review]
Looks great :tada:

**Review by samsondav — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Pull request merged**